src/ray/core_worker/future_resolver.cc

```diff
@@ -88,10 +89,14 @@ void FutureResolver::ProcessResolvedObject(const ObjectID &object_id,
           const_cast<uint8_t *>(reinterpret_cast<const uint8_t *>(metadata.data())),
           metadata.size());
     }
-    auto inlined_ids =
-        IdVectorFromProtobuf<ObjectID>(reply.object().nested_inlined_ids());
-    RAY_UNUSED(in_memory_store_->Put(RayObject(data_buffer, metadata_buffer, inlined_ids),
-                                     object_id));
+    auto inlined_refs =
+        VectorFromProtobuf<rpc::ObjectReference>(reply.object().nested_inlined_refs());
+    for (const auto &inlined_ref : inlined_refs) {
+      reference_counter_->AddBorrowedObject(ObjectID::FromBinary(inlined_ref.object_id()),
```

Comment:
This is the relevant line to fix #17719 (the rest of the PR is plumbing to make sure we pass the owner address around).